src/sidebar/services/autosave.js

```diff
@@ -1,66 +1,70 @@
-/**
- * A service for automatically saving new highlights.
- */
 import { retryPromiseOperation } from '../util/retry';
 
 /**
- * @param {import('./annotations').AnnotationsService} annotationsService
- * @param {import('../store').SidebarStore} store
+ * A service for automatically saving new highlights.
+ *
+ * @inject
  */
-// @inject
-export default function autosaveService(annotationsService, store) {
-  // A Set of annotation $tags that have save requests in-flight
-  const saving = new Set();
-
-  // A Set of annotation $tags that have failed to save after retries
-  const failed = new Set();
-
+export class AutosaveService {
   /**
-   * Determine whether we should try to send a save request for the highlight
-   * indicated by `htag`
-   *
-   * @param {string} htag - The local unique identifier for the unsaved highlight
-   * @return {boolean}
+   * @param {import('./annotations').AnnotationsService} annotationsService
+   * @param {import('../store').SidebarStore} store
    */
-  const shouldSaveHighlight = htag => {
-    return !saving.has(htag) && !failed.has(htag);
-  };
+  constructor(annotationsService, store) {
+    this._store = store;
 
-  /**
-   * Store-subscribed call back. Automatically save new highlights.
-   */
-  const autosaveNewHighlights = () => {
-    const newHighlights = store.newHighlights();
+    // A set of annotation $tags that have save requests in-flight
+    this._saving = new Set();
+
+    // A set of annotation $tags that have failed to save after retries
+    this._failed = new Set();
+
+    /**
+     * Determine whether we should try to send a save request for the highlight
+     * indicated by `htag`
+     *
+     * @param {string} htag - The local unique identifier for the unsaved highlight
+     * @return {boolean}
+     */
+    const shouldSaveHighlight = htag => {
+      return !this._saving.has(htag) && !this._failed.has(htag);
+    };
+
+    /**
+     * Store-subscribed call back. Automatically save new highlights.
+     */
+    this._autosaveNewHighlights = () => {
```

Comment:
I would suggest to either (1) move this function to the `init` method or (2) make it its own method. 

The `shouldSaveHighlight` could be a helper function inside `init` or make it its own method.